Source/JavaScriptCore/runtime/ISO8601.h

```diff
@@ -170,6 +170,43 @@ class ExactTime {
     Int128 m_epochNanoseconds { };
 };
 
+// https://tc39.es/proposal-temporal/#sec-temporal-internal-duration-records
+// Represents a duration as an ISO8601::Duration (in which all time fields
+// are ignored) along with an Int128 time duration that represents the sum
+// of all time fields. Used to avoid losing precision in intermediate calculations.
+class InternalDuration final {
+public:
+    InternalDuration(Duration d, Int128 t)
+        : m_dateDuration(d), m_time(t) { }
+    InternalDuration()
+        : m_dateDuration(Duration()), m_time(0) { }
+    static constexpr Int128 maxTimeDuration = 9'007'199'254'740'992 * ExactTime::nsPerSecond - 1;
+
+    int32_t sign() const;
+
+    int32_t timeDurationSign() const
+    {
+        return m_time < 0 ? -1 : m_time > 0 ? 1 : 0;
+    }
+
+    Int128 time() const { return m_time; }
+
+    Duration dateDuration() const { return m_dateDuration; }
+
+    static InternalDuration combineDateAndTimeDuration(JSGlobalObject*, Duration, Int128);
+private:
+
+    // Time fields are ignored
+    Duration m_dateDuration;
+
+/* A time duration is an integer in the inclusive interval from -maxTimeDuration
+   to maxTimeDuration, where
+   maxTimeDuration = 2**53 × 10**9 - 1 = 9,007,199,254,740,991,999,999,999.
+   It represents the portion of a Temporal.Duration object that deals with time
+   units, but as a combined value of total nanoseconds.  */
```

Comment:
I think it would be better to use line style comment (`// foo`) here for coding style consistency.

Reply:
Done in [aed29bd](https://github.com/WebKit/WebKit/pull/43980/commits/aed29bd52829c0a813d98fdca0332fab2e82e87d)